http/rest-client/src/main/java/io/quarkus/ts/http/restclient/GzipResource.java

```diff
@@ -0,0 +1,22 @@
+package io.quarkus.ts.http.restclient;
+
+import jakarta.inject.Inject;
+import jakarta.ws.rs.POST;
+import jakarta.ws.rs.Path;
+
+import org.eclipse.microprofile.rest.client.inject.RestClient;
+import org.jboss.resteasy.annotations.GZIP;
+
+@Path("/gzip")
+public class GzipResource {
+
+    @Inject
+    @RestClient
+    GzipClientService gzipClient;
+
+    @POST
+    public String gzip(@GZIP byte[] message) {
+        return gzipClient.gzip(message);
```

Comment:
This in just recursive call for itself, this is causing issue reported in https://github.com/quarkusio/quarkus/issues/40806

Reply:
Just do something like 
```
    public String gzip(@GZIP byte[] message) {
        return "OKi";
    }
```